core/src/main/java/org/elasticsearch/common/unit/Fuzziness.java

```diff
@@ -60,6 +65,19 @@ private Fuzziness(String fuzziness) {
         this.fuzziness = fuzziness.toUpperCase(Locale.ROOT);
     }
 
+    private Fuzziness(String fuzziness, int lowDistance, int highDistance) {
+        if (fuzziness == null || fuzziness.isEmpty()) {
+            throw new IllegalArgumentException("fuzziness can't be null!");
+        }
+        if (lowDistance < 0 || highDistance < 0 || lowDistance >= highDistance ) {
+            throw new IllegalArgumentException("fuzziness wrongly configured, must be: lowDistance > 0, highDistance" +
+                " > 0 and lowDistance < highDistance ");
+        }
```

Comment:
should we allow lowDistance==highDistance in case there is no length that is worth using a distance of 1?